allennlp/nn/util.py

```diff
@@ -586,10 +651,21 @@ def get_range_vector(size: int, is_cuda: bool) -> torch.Tensor:
         indices = torch.arange(0, size).long()
     return Variable(indices, requires_grad=False)
 
+
 def bucket_distance(distances: torch.Tensor) -> torch.Tensor:
```

Comment:
I'd probably just call this `bucket_values`, not `bucket_distances` - the caller can use it for distances, but also for other things.  It also seems pretty simple to let the caller customize both the number of identity buckets and the total number of buckets (just moving the `4` and `9` below into parameters with default values here).